meetings/2021-12/dec-14.md

```diff
@@ -0,0 +1,932 @@
+
+MB: Yeah, we will have some work to do on addressing the feedback. But yeah, once we go through all that and you get the official sign-off in the end, we can ask for Stage 3 at a future TC39 meeting. And that's it.
+
+MS: What's your time frame for having reviews done?
+
+MB: Before the next meeting if possible at all would be really nice. The sooner the better as always. If any of the reviewers have time constraints, please let us know so we can plan accordingly.
+
+MS: Okay.
+
+YSV: yeah, so I wanted to mention our position (Mozilla) which is we won't formally block, but we aren't entirely convinced with the solution to add a new flag specifically. We are convinced by like generally, what's being done with the feature and all the work that you folks have done with this. I've been watching this for a while and it has been difficult. But with regards to the flag, we have our reservations and we hope that those won't be borne out. So, not blocking, just raising our concerns.
+
+MB: I think that's it. Happy to move on. Thanks everyone.
+
+### Conclusion/Resolution
+
+- proposal is ready for stage 3 reviewers
+- stated non-blocking concerns about the necessity of a new flag from Mozilla
+
+## Array Grouping for Stage 3
+
+Presenter: Justin Ridgewell (JRL)
+
+- [proposal](https://github.com/tc39/proposal-array-grouping)
+- [criteria](https://github.com/tc39/proposal-array-grouping/issues/19)
+
+JRL: This is super easy because there are essentially no changes. We reached the stage 3 criteria since the last meeting for Array `groupBy` and `groupByMap`. We have spec text. That's been pushed out, reviewers have all approved and editors have all approved - except one change that's currently in flight, but it's an editorial change to the methods.
+
+JRL: The only discussion point which needs to be brought up is, Jordan brought up a groupByMap naming issue. Essentially. He's equating flatMap meaning `map` followed by `flat` and if that would be confusing for people who see groupByMap as map followed by groupBy, or groupBy into a  map output. I'm not super sold on it. I think it's okay to have groupByMap return a map. But that's the only discussion point we have - and then I can ask for stage 3.
+
+JHD: Yeah, I just want to make sure we discussed this. So yeah, I mean it's a relatively minor issue. The polyfill that I already made would have to be renamed so it's not like - this will cause friction for me to rename it anyway, but I still wanted to bring it up. The other thing is - it's kind of unfortunate, right? We have this naming conflict between mapping and a Map and it seems - I don't really have a better suggestion for the name, but it seems like not a great name for the groups, where the groups are maps instead of objects. Before stage 3 is the time to discuss it. So I wanted to bring it up before the advancement.
```

Comment:
```suggestion
JHD: Yeah, I just want to make sure we discussed this. So yeah, I mean it's a relatively minor issue. The polyfill that I already made would have to be renamed so this will cause friction for me to rename it anyway, but I still wanted to bring it up. The other thing is - it's kind of unfortunate, right? We have this naming conflict between mapping and a Map and it seems - I don't really have a better suggestion for the name, but it seems like not a great name for the groups, where the groups are Maps instead of Objects. Before stage 3 is the time to discuss it. So I wanted to bring it up before the advancement.
```